core/shared/src/main/scala/scalaz/zio/Fiber.scala

```diff
@@ -46,12 +46,31 @@ trait Fiber[+E, +A] { self =>
   final def zipWith[E1 >: E, B, C](that: => Fiber[E1, B])(f: (A, B) => C): Fiber[E1, C] =
     new Fiber[E1, C] {
       def join: IO[E1, C] =
-        self.join.zipWith(that.join)(f)
+        self.join.seqWith(that.join)(f)
 
       def interrupt(t: Throwable): IO[Nothing, Unit] =
         self.interrupt(t) *> that.interrupt(t)
     }
 
+  /**
+   * Zips this fiber and the specified fiber togther, producing a tuple of their
+   * output.
+   */
+  final def zip[E1 >: E, B](that: => Fiber[E1, B]): Fiber[E1, (A, B)] =
```

Comment:
This could be changed to `seq` and also, `zipWith` could be changed to `seqWith`.

Reply:
Why? It seems we have just this way of zipping `Fiber`s, we don't have the duality issue we have in `IO`

Reply:
but the name clearly indicates that this is a sequential composition -- it might not be immediately obvious, since we're dealing with thread-like things

Reply:
The "sequential" part occurs in both join and interrupt: `a.seq(b)` means that `a` will be joined first, followed by `b`. It also means that `a` will be interrupted first, followed by `b`.

But I'm not super persuaded either way.